cmd/discovery-server/app/options/options.go

```diff
@@ -54,7 +54,7 @@ func (o *ServingOptions) Validate() []error {
 
 // ApplyTo applies the options to the configuration.
 func (o *ServingOptions) ApplyTo(c *ServingConfig) error {
-	c.Address = net.JoinHostPort(o.Address, strconv.Itoa(int(o.Port)))
+	c.Address = net.JoinHostPort(o.Address, strconv.FormatUint(uint64(o.Port), 10))
```

Comment:
@dimityrmirchev please take a look on this. 
It could be revisited when https://github.com/moby/moby/issues/48358 is fixed.

Reply:
ping @dimityrmirchev